alchemiscale/storage/statestore.py

```diff
@@ -779,49 +826,138 @@ def queue_taskqueue_tasks(
         AlchemicalNetwork queues, or none at all.
 
         If this Task has an EXTENDS relationship to another Task, that Task must
-        be 'complete' before this Task can be added to *any* TaskQueue.
+        be 'complete' before this Task can be added to *any* TaskHub.
 
         """
         for t in tasks:
             q = f"""
-            // get our task queue, as well as tail and tail relationship to last in line
-            MATCH (tq:TaskQueue {{_scoped_key: '{taskqueue}'}})-[:TASKQUEUE_TAIL]->
-                      (tqt)-[tqtl:FOLLOWS {{taskqueue: '{taskqueue}'}}]->
-                      (last),
-                  (tq)-[:PERFORMS]->(an:AlchemicalNetwork)
+            // get our TaskHub, 
+            MATCH (th:TaskHub {{_scoped_key: '{taskhub}'}})-[:PERFORMS]->(an:AlchemicalNetwork)
             
             // get the task we want to add to the queue; check that it connects to same network
             // if it has an EXTENDS relationship, get the task it extends
             MATCH (tn:Task {{_scoped_key: '{t}'}})-[:PERFORMS*]->(tf:Transformation)<-[:DEPENDS_ON]-(an)
             OPTIONAL MATCH (tn)-[:EXTENDS]->(other_task:Task)
 
             // only proceed for cases where task is not already in queue and only EXTENDS a 'complete' task
-            WHERE NOT (tqt)-[:FOLLOWS* {{taskqueue: '{taskqueue}'}}]->(tn)
+            WHERE NOT (th)-[:ACTIONS* {{taskhub: '{taskhub}'}}]->(tn)
               AND other_task.status = 'complete'
 
-            // create the connections that add it to the end of the queue, and delete old queue tail connection
-            CREATE (tqt)-[:FOLLOWS {{taskqueue: '{taskqueue}'}}] ->
-                      (tn)-[:FOLLOWS {{taskqueue: '{taskqueue}'}}]->(last)
-            DELETE tqtl
-
+            // create the connections 
+            CREATE (th)-[ar:ACTIONS {{taskhub: '{taskhub}'}}] ->
```

Comment:
Might want to make this a MERGE statement instead of CREATE; should make the whole operation idempotent if run multiple times, avoiding duplicate ACTIONS relationships.

Reply:
Ah, nevermind. I see we do a WHERE up above which should be sufficient to avoid hitting this if the relationship already exists.